js/Learn JavaScript concepts by building a random quote generator.md

```diff
@@ -0,0 +1,289 @@
+```
+So all our quotes are basically strings like this.
+```
+'Stay hungry, stay foolish !'
+```
+ But there are multiple such strings and we need something to store them. Here comes the concept of Arrays. But what's an array?
+
+### Arrays
+
+> 
+An array in JavaScript is a type of global object that is used to store data. Arrays consist of an ordered collection or list containing zero or more data types, and use numbered indices starting from 0 to access specific items.
+ 
+```
+<!--Creating an array-->
+let quotes =[];
+```
+Arrays can contain any data type, including numbers, strings, and objects. We will be storing our strings in an array like this.
+```javascript
+let quotes=[
+'The secret to life is to love who you are.',
+'Look for opportunities in every change in your life.',
+'Persist while others are quitting.',
+'and so on.'
+];
+```
+### Array.length
+The length property of an object which is an instance of type Array sets or returns the number of elements in that array.
+```
+quotes.length;
+<!--Output-->
+4
+```
+
+#### Accessing Items in an Array
+An item in a JavaScript array is accessed by referring to the index number of the item in square brackets.
+```javascript
+quotes[2];
+<!--Output : -->
+Persist while others are quitting.
+
+```
+But here we will not access like this. We want a random index to display a random quote. Then how do we do that? We will use the math object.
+
+### Using Math object
+
+> 
+Math is a built-in object that has properties and methods for mathematical constants and functions.
+
+![Screenshot (242).png](https://cdn.hashnode.com/res/hashnode/image/upload/v1585047447641/KLYjO52E1.png)
+
+If you type math in console, you will see there are so many built-in methods for Math object. We will be using a few of them.
+
+### Math.random() 
+The Math.random() function returns a floating-point, pseudo-random number in the range 0 to less than 1. See how it returns every time we call it in the console.
+
+
+![Screenshot (243).png](https://cdn.hashnode.com/res/hashnode/image/upload/v1585151170351/FFMTnGRBt.png)
+
+
+But this is not what we want. We want a whole number and not a floating type. For this, the Math object provides another method called Math.floor()
+
+### Math.floor()
+Math.floor() returns a number representing the largest integer less than or equal to the specified number. So if we pass the math.random() as a parameter to math.floor() we will get a whole number.
+But Math.floor() is always rounding down to the nearest decimal, therefore, every decimal between 0 and 1 will always revert back to 0.
+
+
+![Screenshot (245).png](https://cdn.hashnode.com/res/hashnode/image/upload/v1585047247928/aNY04WDp2.png)
+
+In order to get a random number between 0 to the length of our array, we will multiply the math.random() by quotes.length. 
+Finally the random index we want is 
+```javascript
+let index=Math.floor(Math.random()*quotes.length);
+<!--This will give a random quote-->
+quotes[index];
+```
```

Comment:
```suggestion
```javascript
let index = Math.floor(Math.random() * quotes.length); // returns a random number that will be stored in our index variable
quotes[index]; // Output: return quote at the `index`
```
```